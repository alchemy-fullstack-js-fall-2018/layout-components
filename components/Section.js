import Component from './Component';
import styles from './Section.css';

export default class Section extends Component {
  render() {
    const sectionContent = this.props.sectionContent;

    return `
        <section class="${styles.section}">
            ${sectionContent}
        </section>`;
  }
}
