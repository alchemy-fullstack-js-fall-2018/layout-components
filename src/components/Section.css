.section {
  font-family: sans-serif;
}

.p {
  background-color: orchid;
  margin: auto;
}