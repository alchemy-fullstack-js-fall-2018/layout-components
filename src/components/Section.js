import Component from './Component';

export default class Section extends Component {
  render() {
    const children = this.props.children.map(child  => {
      return child.render();
    });



    return `
      <section>
        <div>
          ${children}
        </div>
      </section>
    `;
  }
}
