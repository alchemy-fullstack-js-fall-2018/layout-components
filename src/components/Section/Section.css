#container {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
}

.tile {
  margin: 5px;
}

._Raiders {
  background-color: black;
  color: silver;
}

._Patriots {
  background-color: navy;
  color: red;
}

._49ers {
  background-color: red;
  color: gold;
}

._Rams {
  background-color: blue;
  color: gold;
}

._Ravens {
  background-color: purple;
  color: black;
}
