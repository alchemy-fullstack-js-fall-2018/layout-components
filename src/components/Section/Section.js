import Component from '../Component/Component';

import styles from './Section.css';
export default class Section extends Component {

  render() {


    const teamAbout = this.props.teams.map(team => {
      return `
        <div class="${styles.tile} ${styles[`_${team.name}`]}">
          <h2>${team.name}</h2>
          <p>${team.story}</p>
        </div>
      `;
    });

    return `
      <section id="${styles.container}">
        ${teamAbout.join(' ')}
      </section>
    `;
  }


}
