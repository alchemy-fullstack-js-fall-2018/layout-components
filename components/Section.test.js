import Section from './Section';

describe('Section Component', () => {
  it('should be a section', () => {
    const section = new Section({
      sectionContent: 'section'
    });

    expect(section.render()).toMatchSnapshot();
  });
});
