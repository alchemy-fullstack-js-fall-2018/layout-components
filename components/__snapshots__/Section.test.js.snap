// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should be a header 1`] = `
"
        <header class=\\"header\\">
            <nav>
                <a href=\\"/about\\">about</a>
            </nav>
        </header>"
`;

exports[`Section Component should be a header 1`] = `
"
        <section class=\\"section\\">
            undefined
        </section>"
`;

exports[`Section Component should be a section 1`] = `
"
        <section class=\\"section\\">
            undefined
        </section>"
`;
